airflow/docker.go

```diff
@@ -316,6 +323,26 @@ func checkServiceState(serviceState, expectedState string) bool {
 	return scrubbedState == expectedState
 }
 
+func (d *DockerCompose) webserverHealthCheck() error {
+	err := d.composeService.Events(context.Background(), d.projectName, api.EventsOptions{
+		Services: []string{config.CFG.WebserverContainerName.GetString()}, Consumer: func(event api.Event) error {
+			if event.Status == "health_status: healthy" {
+				fmt.Println("\nProject is running! All components are now available.")
+				// have to return an error to break from the event listener loop
+				return errComposeProjectRunning
+			} else if event.Status == "exec_die" {
```

Comment:
The `exec_die` event status is only returned when Airflow components are not up yet ?

If it's also returned during unknown errors, how do you know you should stop the health check and return an error from this function ?

Reply:
`exec_die` event corresponds to ending of any `exec` call, which in this case will be for each health check call invoked by docker (heath check is defined here: https://github.com/astronomer/astro-cli/pull/486/files#diff-3025f0d00360d8168d677ee4b64f6f24b11069b778029c2127a80b50a01d62b3R143-R148).

This function is only involved to check the result of the health check done internally by Docker. States which are usually reported by Docker would be starting, healthy or unhealthy. Docker on its own, doesn't take any action even in case container is unhealthy, it will simply report it and we in this function are just consuming that information to figure out when the container is healthy.

Just added a counter and a break condition to break out of that loop as a fail safe condition.